File: 백준/Silver/11047. 동전 0/동전 0.py
import sys
input = sys.stdin.readline

N, K = map(int, input().split())
A = [0] * N
for _ in range(N):
    A[_] = int(input())

cnt = 0

for i in range(N-1, -1, -1):
    if A[i] > K:
        continue

    if K == 0:
        break

    n, K = divmod(K, A[i])
    cnt += n

print(cnt)

File: 백준/Silver/11047. 동전 0/동전 0.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

let [n, k] = input[0].split(' ').map(Number);
let coins = [];

for (let i = 1; i <= n; i++) {
  coins.push(Number(input[i]));
}
coins.reverse();

let answer = 0;
for (const coin of coins) {
  answer += Math.floor(k / coin);
  k %= coin;
}

console.log(answer);
